source/distributed/tria.cc

```diff
@@ -3491,32 +3491,42 @@ namespace parallel
       typename std::map<std::pair<cell_iterator, unsigned int>,
                std::pair<std::pair<cell_iterator,unsigned int>, std::bitset<3> > >::const_iterator it;
 
-      for (it = this->get_periodic_face_map().begin(); it!= this->get_periodic_face_map().end(); ++it)
-        {
-          const cell_iterator &cell_1 = it->first.first;
-          const unsigned int face_no_1 = it->first.second;
-          const cell_iterator &cell_2 = it->second.first.first;
-          const unsigned int face_no_2 = it->second.first.second;
-          const std::bitset<3> &face_orientation = it->second.second;
+      // When a connectivity in the code below is detected, the assignment
+      // 'marked_vertices[v1] = marked_vertices[v2] = true' makes sure that
+      // the information about the periodicity propagates back to vertices on
+      // cells that are not owned locally. However, in the worst case we want
+      // to connect to a vertex that is 'dim' hops away from the locally owned
+      // cell. Depending on the order of the periodic face map, we might
+      // connect to that point by chance or miss it. However, after looping
+      // through the faces once per space dimension we can be sure that all
+      // connections to vertices have been made.
+      for (unsigned int i=0; i<dim; ++i)
```

Comment:
can you use a different variable name than ``i``? Maybe ``repetition`` or something?